src/extension/tab/hook.ts

```diff
@@ -97,7 +97,16 @@ function initializeHook() {
       payload: {
         queries: hook.getQueries(),
         mutations: hook.getMutations(),
-        cache: hook.getCache(),
+
+        // We need to JSON stringify the cache here in case the cache contains
+        // references to `URL` instances which are not cloneable via
+        // `structuredClone` (which `window.postMessage` uses to send messages).
+        // `JSON.stringify` does however serialize `URL`s into strings properly,
+        // so this should ensure that the cache data will be sent without
+        // errors.
+        //
+        // https://github.com/apollographql/apollo-client-devtools/issues/1258
+        cache: JSON.stringify(hook.getCache()),
```

Comment:
doesn't seem that this fully fixed the issue. Since the payload includes queries, mutations, and a cache, and each of them could contain data that [doesn't work with structuredClone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#things_that_dont_work_with_structured_clone), shouldn't the whole thing be stringified? 
```ts
function sendHookDataToDevTools(eventName: "update" | "connectToDevtools") {
    tab.send({
      type: eventName,
      payload: JSON.stringify({
        queries: hook.getQueries(),
        mutations: hook.getMutations(),
        cache: hook.getCache(),
      }),
    });
  }
```

Reply:
Argh, you're right. I missed that `queries` and `mutations` have [`cachedData`](https://github.com/apollographql/apollo-client-devtools/blob/df5300f245b82e6efc15f616724874fbbb0b8a2d/src/extension/tab/helpers.ts#L14) on them. The `DocumentNode` and `variables` should always be fine otherwise. 

Thanks for pointing this out! I'll get another quick fix in place.

Reply:
This should be fixed with https://github.com/apollographql/apollo-client-devtools/pull/1268

Reply:
The above fix was released in 4.8.3 and should hit the web stores any time now. Thanks again for pointing this out!